modin/core/dataframe/algebra/binary.py

```diff
@@ -424,4 +426,5 @@ def caller(
                     new_modin_frame, shape_hint=shape_hint
                 )
 
+        cls._binary_ops.add(func)
```

Comment:
Why should we put a function for all binary cases? I guess this PR only addresses map case, doesn't it?

Reply:
I think, it makes sense to use the lazy execution for all the binary functions.

Reply:
Then I think we would need to look at perf measurements for all cases of binary operations (aligned frames, unaligned frames, etc.), which may take time to investigate this and do not promise positive results. I would propose to leave this PR only for the map case of binary operation.

Reply:
Done.